packages/datastore/src/datastore.ts

```diff
@@ -34,13 +42,35 @@ import {
  * https://hal.inria.fr/file/index/docid/555588/filename/techreport.pdf
  */
 export
-class Datastore implements IIterable<Table> {
+class Datastore implements IIterable<Table<Schema>> {
+
+  /**
+   * Create a new datastore.
+   *
+   * @param id - The unique id of the datastore.
+   * @param schemas - The table schemas of the datastore.
+   */
+  constructor(id: number, schemas: ReadonlyArray<Schema>) {
+    this._context = {
+      inTransaction: false,
+      version: 0,
+      storeId: id,
+      change: {},
+      patch: {},
+    };
+    this._tables = {};
+    for (let s of schemas) {
+      this._tables[s.id] = Table.create(s, this._context);
+    }
```

Comment:
You'll want to validate the schemas to ensure the field names don't violate the rules listed in the documentation.

Since you may need to throw an exception (and it's bad practice to throw from constructors), you might want to make the constructor private, and define a `create` static method on the class which does the construction.